CRM/Event/Form/Participant.php

```diff
@@ -1618,14 +1603,17 @@ protected function addParticipant($contactID) {
       'role_id' => $this->getSubmittedValue('role_id'),
       'register_date' => $this->getSubmittedValue('register_date'),
       'source' => $this->getSourceText(),
-      'fee_level' => $this->order->getAmountLevel(),
       'is_pay_later' => FALSE,
-      'fee_amount' => $this->order->getTotalAmount(),
       'fee_currency' => $this->getCurrency(),
       'campaign_id' => $this->getSubmittedValue('campaign_id'),
       'note' => $this->getSubmittedValue('note'),
-      'is_test' => ($this->_mode === 'test)'),
+      'is_test' => $this->isTest(),
     ];
+    $order = $this->getOrder();
+    if ($order) {
+      $participantParams['fee_level'] = $this->order->getAmountLevel();
```

Comment:
This doesn't look quite right compared to L1615

Reply:
Yeah true - it would work fine but it's inconsistent

Reply:
I fixed it